### PR TITLE
Integrate buttons into logs

### PR DESCRIPTION
**Pull request by curled-bash**
- adds view to log message
- ban user and unbans user and links to context
- ***SHOULD BE MERGED ONLY AFTER #5 IS CHECKED OFF!***

### CODE REVIEW

File: src/confess.py [hunk 1]
```diff
@@ -41,17 +53,28 @@ async def confess(
     # TODO use embed.set_footer to set IDs
     # TODO set up moderation logging
 
-    message = await interaction.channel.send(embed=embed)
-
+    # TODO figure out how to do these simultaneously
+    confession_msg: discord.Message = await interaction.channel.send(embed=embed)
     await interaction.response.send_message("Confession sent!", ephemeral=True)
```
Comment (emmajsadams) [inline src/confess.py:58]: I think you can handle this by including `embed=embed` in the parameters of this method call. Seems like send_message supports this api https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.InteractionResponse.send_message

File: src/database.py [hunk 2]
```diff
@@ -71,17 +71,56 @@ def log_message(message_id: int, author_id: int, created_at: datetime.datetime)
...
-def log_message_from_message(message: discord.Message) -> None:
+def log_message_from_message(message: discord.Message, author_id: int) -> None:
     message_id = message.id
-    author_id = message.author.id
     created_at: datetime.datetime = message.created_at
     log_message(message_id, author_id, created_at)
 
 
-def add_ban_log(user_id: int) -> None:
+# TODO figure out if we want to use a class or unnecessary overhead
```
Comment (emmajsadams) [inline src/database.py:80]: not quite sure I understand this todo? or what the trade off is? Regardless this code seems fine.

File: src/ban.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+@client.tree.command(description="Ban a user from confessions")
+@app_commands.describe(
+    sequential_id='id of confession of author to be banned',
+)
+async def ban_user(
+        interaction: discord.Interaction,
+        sequential_id: int,
+):
+    success = database.add_ban_log_by_sequential_id(sequential_id)
```
Comment (emmajsadams) [inline src/ban.py:16]: so `sequential_id` refers to our internal SQLITE db for a confession right? And discord is calling this function and passing our id into it?

**Review by emmajsadams — APPROVED** [review]
lgtm didn't pull this down not enough time today, but no need to hold these prs up on me!

**Comment by emmajsadams** [comment]
Oh it seems like PR #6 depends on #5 right? One way of handling this in the future is you can branch #6 off #5. Then merge #5 first, pull master into #6, and that makes handling the merges a little easier as well as makes each review focus on what is unique in each review. Since right now it seems like #5 contains some of #6 also.

**Comment by curled-bash** [comment]
got it, that's actually a much better way to deal with that, thank you :)

**Pull request merged**